addons/api/addon/serializers/credential-library.js

```diff
@@ -14,16 +17,41 @@ export default class CredentialLibrarySerializer extends ApplicationSerializer {
    * @method serialize
    * @param {Snapshot} snapshot
    */
-  serialize() {
+  serialize(snapshot) {
+    switch (snapshot.record.type) {
+      case TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC:
+        return this.serializeVaultGeneric(...arguments);
+      case TYPE_CREDENTIAL_LIBRARY_VAULT_SSH_CERTIFICATE:
+      default:
+        return super.serialize(...arguments);
+    }
+  }
+
+  serializeAttribute(snapshot, json, key, attribute) {
+    const value = super.serializeAttribute(...arguments);
+    const { options } = attribute;
+    const { type } = snapshot.record;
+
+    // For any attribute that doesn't match its `for`
+    // or isn't undefined, we delete it from the json
+    if (
+      options.isNestedAttribute &&
+      options.for !== type &&
+      options.for !== undefined
+    ) {
+      delete json.attributes[key];
+    }
+
+    return value;
```

Comment:
This is just a general question about the `serializeAttribute` method. I don't understand why there is a `return value` since [the docs](https://api.emberjs.com/ember-data/4.9/classes/JSONAPISerializer/methods/serializeAttribute?anchor=serializeAttribute) don't specify the method returning anything.

Reply:
Yes you're right, there is no return value so it doesn't do anything. I admittedly was following a previous example we had but I can clean this up